tests/k8st/tests/test_bgp_advert.py

```diff
@@ -165,7 +165,42 @@ def test_mainline(self):
         retry_until_success(lambda: self.assertIn(local_svc_ip, self.get_routes()))
         retry_until_success(lambda: self.assertNotIn(cluster_svc_ip, self.get_routes()))
 
-        # Delete nginx service
+        # Create a network policy that only accepts traffic from the external node
+        run("""kubectl exec -i -n kube-system calicoctl -- /calicoctl apply -f - << EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-tcp-80-ex
+  namespace: bgp-test
+spec:
+  selector: all()
+  types:
+  - Ingress
+  ingress:
+  - action: Allow
+    protocol: TCP
+    source: { nets: ["10.192.0.5/32"] }
+    destination: { ports: ["80"] }
+EOF
+""")
+
+        # Connectivity to nginx-local should always succeed
+        self.do_external_curl("kube-node-extra", local_svc_ip)
+
+        # Connectivity to nginx-cluster will rarely succeed, try to curl 10 times
+        try:
+          for i in range(10):
+            self.do_external_curl("kube-node-extra", cluster_svc_ip)
+          self.fail("external node should not be able to access cluster svc")
+        except subprocess.CalledProcessError:
+          pass
+
+        # Scale the local_svc to 2 replicas
```

Comment:
I'd recommend doing 3-4 replicas, just to ensure there's a replica on more than one node (I'd be worried about two ending up on the same node for some reason).